spec.html

```diff
@@ -1585,158 +1585,121 @@ <h1>Number::bitwiseNOT ( _x_ )</h1>
-            </li>
-          </ul>
+          <p>The abstract operation Number::divide takes arguments _x_ (a Number) and _y_ (a Number). It performs division, producing the quotient of _x_ and _y_ where _x_ is the dividend and _y_ is the divisor. It performs the following steps when called:</p>
+          <emu-alg>
+            1. If _x_ is *NaN* or _y_ is *NaN*, return *NaN*.
+            1. If _x_ is *+&infin;*<sub>𝔽</sub> or _x_ is *-&infin;*<sub>𝔽</sub>, then
+              1. If _y_ is *+&infin;*<sub>𝔽</sub> or _y_ is *-&infin;*<sub>𝔽</sub>, return *NaN*.
+              1. If _y_ is *+0*<sub>𝔽</sub> or _y_ is positive, return _x_.
+              1. If _x_ is *+&infin;*<sub>𝔽</sub>, return *-&infin;*<sub>𝔽</sub>. Otherwise, return *+&infin;*<sub>𝔽</sub>.
+            1. If _y_ is *+&infin;*<sub>𝔽</sub>, then
+              1. If _x_ is *+0*<sub>𝔽</sub> or _x_ is positive, return *+0*<sub>𝔽</sub>. Otherwise, return *-0*<sub>𝔽</sub>.
+            1. If _y_ is *-&infin;*<sub>𝔽</sub>, then
+              1. If _x_ is *+0*<sub>𝔽</sub> or _x_ is positive, return *-0*<sub>𝔽</sub>. Otherwise, return *+0*<sub>𝔽</sub>.
+            1. If _x_ is *+0*<sub>𝔽</sub> or _x_ is *-0*<sub>𝔽</sub>, then
+              1. If _y_ is *+0*<sub>𝔽</sub> or _y_ is *-0*<sub>𝔽</sub>, return *NaN*.
+              1. If _y_ is positive, return _x_.
+              1. If _x_ is *+0*<sub>𝔽</sub>, return *-0*<sub>𝔽</sub>. Otherwise, return *+0*<sub>𝔽</sub>.
+            1. If _y_ is *+0*<sub>𝔽</sub>, then
+              1. If _x_ is positive, return *+&infin;*<sub>𝔽</sub>. Otherwise, return *-&infin;*<sub>𝔽</sub>.
+            1. If _y_ is *-0*<sub>𝔽</sub>, then
+              1. If _x_ is positive, return *-&infin;*<sub>𝔽</sub>. Otherwise, return *+&infin;*<sub>𝔽</sub>.
+            1. Return 𝔽(ℝ(_x_) / ℝ(_y_)).
+          </emu-alg>
         </emu-clause>
 
         <emu-clause id="sec-numeric-types-number-remainder" oldids="sec-applying-the-mod-operator">
           <h1>Number::remainder ( _n_, _d_ )</h1>
-          <p>The abstract operation Number::remainder takes arguments _n_ (a Number) and _d_ (a Number). It yields the remainder of its operands from an implied division; _n_ is the dividend and _d_ is the divisor.</p>
+          <p>The abstract operation Number::remainder takes arguments _n_ (a Number) and _d_ (a Number). It yields the remainder from an implied division of its operands where _n_ is the dividend and _d_ is the divisor. It performs the following steps when called:</p>
+          <emu-alg>
+            1. If _n_ is *NaN* or _d_ is *NaN*, return *NaN*.
+            1. If _n_ is *+&infin;*<sub>𝔽</sub> or _n_ is *-&infin;*<sub>𝔽</sub>, return *NaN*.
+            1. If _d_ is *+0*<sub>𝔽</sub> or _d_ is *-0*<sub>𝔽</sub>, return *NaN*.
+            1. If _d_ is *+&infin;*<sub>𝔽</sub> or _d_ is *-&infin;*<sub>𝔽</sub>, return _n_.
+            1. If _n_ is *+0*<sub>𝔽</sub> or _n_ is *-0*<sub>𝔽</sub>, return _n_.
```

Comment:
Nit: if you switch this line and the previous, then we check in the order n, d, n, d, instead of (as here) n, d, d, n.

Reply:
I did a little more swapping here to group the infinity/0 cases. I *think* it was okay to do that, but double check for me.

Reply:
No, it's wrong now: this now says that `0 % 0` is `0` instead of (as it should be) `NaN`. The check for `d = 0` needs to precede the check for `n = 0`.